rbtcwrapperproxy/RBTCWrapperProxy.sol

```diff
@@ -206,30 +206,35 @@ contract RBTCWrapperProxy is ContractRegistryClient {
 
         ILiquidityPoolV1Converter _liquidityPoolConverter = ILiquidityPoolV1Converter(_liquidityPoolConverterAddress);
         ISmartToken _poolToken = ISmartToken(address(_liquidityPoolConverter.token()));
+        uint256 totalSupplyBefore = _poolToken.totalSupply();
+        uint32 reserveRatio_ = _liquidityPoolConverter.reserveRatio();
+
         uint256 poolTokenAmountBefore = _poolToken.balanceOf(address(this));
         _liquidityPoolConverter.addLiquidity(_reserveTokens, _reserveAmounts, _minReturn);
         uint256 poolTokenAmount = _poolToken.balanceOf(address(this)).sub(poolTokenAmountBefore);
         
         success = _poolToken.transfer(msg.sender, poolTokenAmount);
         require(success, "Failed to transfer pool token to user");
 
-        (uint256 rsvBalance, , , , ) = _liquidityPoolConverter.reserves(wrbtcTokenAddress);
-        uint256 costReserveAmount = ISovrynSwapFormula(addressOf(SOVRYNSWAP_FORMULA)).fundCost(_poolToken.totalSupply(), rsvBalance, _liquidityPoolConverter.reserveRatio(), poolTokenAmount);
-        if (_reserveAmounts[0] > costReserveAmount) {
-            IWrbtcERC20(wrbtcTokenAddress).withdraw(_reserveAmounts[0].sub(costReserveAmount));
-            (success,) = msg.sender.call.value(_reserveAmounts[0].sub(costReserveAmount))("");
-            require(success, "Failed to send extra RBTC back to user");
-        }
         for (uint256 i = 1; i < _reserveTokens.length; i++) {
-            (rsvBalance, , , , ) = (_liquidityPoolConverter.reserves(address(_reserveTokens[i])));
-            costReserveAmount = ISovrynSwapFormula(addressOf(SOVRYNSWAP_FORMULA)).fundCost(_poolToken.totalSupply(), rsvBalance, _liquidityPoolConverter.reserveRatio(), poolTokenAmount);
-            if (_reserveAmounts[i] > costReserveAmount) {
-                uint256 amount = _reserveAmounts[i].sub(costReserveAmount);
-                success = _reserveTokens[i].transfer(msg.sender, amount);
+            (uint256 rsvBalance, , , , ) = _liquidityPoolConverter.reserves(address(_reserveTokens[i]));
+            uint256 costReserveAmount = ISovrynSwapFormula(addressOf(SOVRYNSWAP_FORMULA)).fundCost(totalSupplyBefore, rsvBalance, reserveRatio_, poolTokenAmount);
```

Comment:
I think `ISovrynSwapFormula.fundCost` should use `rsvBalance` calculated before `addLiquidity`
if so, this code overestimates `costReserveAmount` and returns a bit less tokens to the user